server-spi-private/src/main/java/org/keycloak/broker/provider/util/SimpleHttp.java

```diff
@@ -129,6 +126,14 @@ public static SimpleHttp doPatch(String url, HttpClient client) {
         return new SimpleHttp(url, "PATCH", client);
     }
 
+    public String method() {
```

Comment:
I'd say that SimpleHttp is kind of a ResponseBuilder, but it also encapsulates the request info, which is not included in the "builder result class" `SimpleHttp.Response` or in any intermediary Request dto - making SimpleHttp effectively the Request representation itself.
Since SimpleHttp is passed around in the class hierarchy the site of construction/building of SimpleHttp is not under control of the `AbstractOAuth2IdentityProvider` operating on the instance, does it not make sense to be able to inspect, fully immutably, the request at the execution site?

Reply:
+1

I see that you are using methods like `method()`, `url()` and `parameters()` just for the logging. So how about overriding method `toString()` on `SimpleHttp` instead?